draft-ietf-quic-http.md

```diff
@@ -925,7 +925,9 @@ transferred. Endpoints MUST NOT consider these streams to have any meaning upon
 receipt.
 
 The payload and length of the stream are selected in any manner the
-implementation chooses.
+implementation chooses.  Implementations MAY terminate these streams cleanly,
+or MAY abruptly terminate them with an error code of the implementation's
+choice, including reserved error codes ({{http-error-codes}}).
```

Comment:
This allows error codes here that might tear down the entire connection – did you mean to allow that?

Reply:
I dont think that's the case. Closing a grease stream cleanly or abruptly should not be any cause for an H3 level connection error

Reply:
I agree, but this text allows for an implementation to use any error code. The peer could be wired to close the connection on some of these errors, irrespective of stream (H3_CLOSED_CRITICAL_STREAM perhaps). It might be worth stating that the endpoint MUST NOT close the connection on any error on these streams.

Reply:
I guess you could add that specific clause but we already have a MUST NOT consider these streams to have any meaning. 

Zooming out, this is a slight retread of the "what should I do if an endpoint sends me an unexpected error code?" Q
question. Which I thought had been answered.

Reply:
The streams may not have meaning, but does that include the errors?

Reply:
We already say https://quicwg.org/base-drafts/draft-ietf-quic-http.html#section-8

> Because new error codes can be defined without negotiation (see Section 9), use of an error code in an unexpected context or receipt of an unknown error code MUST be treated as equivalent to H3_NO_ERROR. However, closing a stream can have other effects regardless of the error code

Closing a stream with the error code H3_CLOSED_CRITICAL_STREAM seems like an unexpected context to me. Martin makes the suggestion above to also treat reserved error values as H3_NO_ERROR.

By defining and exercising grease of error codes, we can help surface naiive implementations that react in the unfortunate way of your example.

Reply:
As @LPardue says, we only have defined semantics for CLOSED_CRITICAL_STREAM as a connection error.  (And, for that matter, those semantics don't include any action on the receiver's part.)  Receiving it on a stream -- especially a grease stream -- is supposed to have no effect.  If it does, greasing is working.